archivebox/search/backends/sqlite.py

```diff
@@ -0,0 +1,164 @@
+import codecs
+from typing import List, Optional, Generator
+import sqlite3
+
+from archivebox.util import enforce_types
+from archivebox.config import (
+    FTS_SEPARATE_DATABASE,
+    FTS_TOKENIZERS,
+    FTS_SQLITE_MAX_LENGTH
+)
+
+FTS_TABLE = "snapshot_fts"
+FTS_ID_TABLE = "snapshot_id_fts"
+FTS_COLUMN = "texts"
+
+if FTS_SEPARATE_DATABASE:
+    database = sqlite3.connect("search.sqlite3")
+    # Make connection callable, because `django.db.connection.cursor()`
+    # has to be called to get a context manager, but sqlite3.Connection
+    # is a context manager without being called.
+    def connection():
+        return database
+    SQLITE_BIND = "?"
+else:
+    from django.db import connection as database
+    connection = database.cursor
```

Comment:
Do you mean this line, when you say monkey patching? `connection` is a module-local symbol that doesn't affect `django.db`. Is there somewhere else I've fiddled with outside modules, or does the comment above just make it sound that way?

Reply:
that's the line I meant, thanks for clarifying

I think I misread it as doing `django.db.connection = database.cursor` instead of just modifying module level local scope `connection = ...`.